src/Uno.UI/UI/Xaml/Controls/Image/Image.wasm.cs

```diff
@@ -90,121 +90,53 @@ public ImageSource Source
 			set => SetValue(SourceProperty, value);
 		}
 
-		// Using a DependencyProperty as the backing store for Source.  This enables animation, styling, binding, etc...
 		public static readonly DependencyProperty SourceProperty =
 			DependencyProperty.Register("Source", typeof(ImageSource), typeof(Image), new PropertyMetadata(null, (s, e) => ((Image)s)?.OnSourceChanged(e)));
 
 		private void OnSourceChanged(DependencyPropertyChangedEventArgs e)
 		{
 			UpdateHitTest();
 
-			var source = e.NewValue as ImageSource;
-
 			_lastMeasuredSize = _zeroSize;
 
-			var stream = source?.Stream;
-			if (stream != null)
-			{
-				stream.Position = 0;
-				var encodedBytes = Convert.ToBase64String(stream.ReadBytes());
-				var url = "data:application/octet-stream;base64," + encodedBytes;
-				SetImageUrl(url);
-			}
-			else if (source is WriteableBitmap wb)
-			{
-				void setImageContent()
-				{
-					if (wb.PixelBuffer is InMemoryBuffer mb)
-					{
-						var gch = GCHandle.Alloc(mb.Data, GCHandleType.Pinned);
-						var pinnedData = gch.AddrOfPinnedObject();
-
-						try
-						{
-							WebAssemblyRuntime.InvokeJS(
-								"Uno.UI.WindowManager.current.setImageRawData(" + _htmlImage.HtmlId + ", " + pinnedData + ", " + wb.PixelWidth + ", " + wb.PixelHeight + ");"
-							);
-
-							InvalidateMeasure();
-						}
-						finally
-						{
-							gch.Free();
-						}
-					}
-				}
-
-				void OnInvalidated(object sdn, EventArgs args)
-				{
-					setImageContent();
-				}
-
-				wb.Invalidated += OnInvalidated;
-				_sourceDisposable.Disposable = Disposable.Create(() => wb.Invalidated -= OnInvalidated);
-				setImageContent();
-			}
-			else
+			if (e.NewValue is ImageSource source)
 			{
-				void setImageContent()
+				_sourceDisposable.Disposable = source.Subscribe(img =>
 				{
-					var url = source?.WebUri;
-
-					if (url != null)
+					switch (img.Kind)
 					{
-						if (url.IsAbsoluteUri)
-						{
-							if (url.Scheme.Equals("file", StringComparison.OrdinalIgnoreCase))
+						case ImageDataKind.Empty:
+							_htmlImage.SetAttribute("src", "");
+							break;
+
+						case ImageDataKind.Base64:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty 'case' clause.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5633443)